### PR TITLE
fix: room status race condition

### PR DESCRIPTION
**Pull request by dextertanyj**

### CODE REVIEW

File: backend/src/room/room.service.ts [hunk 1]
```diff
@@ -289,9 +304,18 @@ export class RoomService
...
     }
-    return members.map((info) => ({
-      userId: Number(info.split(RoomService.DELIMITER)[0]),
-      isConnected: Boolean(Number(info.split(RoomService.DELIMITER)[1])),
+    const map = new Map<number, boolean>();
+    for (const member of members) {
+      const userId = Number(member.split(RoomService.DELIMITER)[0]);
+      const isConnected =
+        map.get(userId) ||
+        false ||
```
Comment (ypinhsuan) [inline backend/src/room/room.service.ts:312]:
```suggestion
```

File: backend/src/room/room.service.ts [hunk 2]
```diff
@@ -225,20 +229,31 @@ export class RoomService
...
       )
       .deleteFromSet(
         [RoomService.NAMESPACE, RoomService.MEMBERS_NAMESPACE],
         roomId,
-        `${userId.toString()}:${Status.CONNECTED}`,
+        `${userId.toString()}${RoomService.DELIMITER}${socketId}`,
       )
       .execute();
+    const members = await this.getMembers(roomId);
+    console.log(members);
```
Comment (ypinhsuan) [inline backend/src/room/room.service.ts:253]: Maybe remove this or use the logger?

**Review by ypinhsuan — APPROVED** [review]
LGTM

File: backend/src/room/room.service.ts (hunk 1 above)
Comment (dextertanyj) [reply backend/src/room/room.service.ts:312]: Need to convert the undefined to false 🥲

Comment (ypinhsuan) [reply backend/src/room/room.service.ts:312]: Oh oops I thought it wasn't needed cuz there is a boolean expression behind 😮

Comment (dextertanyj) [reply backend/src/room/room.service.ts:312]: Oh right that part totally flew over my head.

**Pull request merged**